### PR TITLE
Update request_time to include additional metrics

### PR DESCRIPTION
**Pull request by kobuskc**
I needed more detail on the request_time metric and would like to propose these changes. Most important to note is that In order to avoid inconsistent label cardinality, I made this change: https://github.com/gesellix/couchdb-prometheus-exporter/compare/master...kobuskc:v2_request_time?expand=1#diff-655827a3ba50a94c5273e509e0750ab9

I cannot see that it will break any existing configuration though.

### CODE REVIEW

File: lib/exporter.go [hunk 1]
```diff
@@ -55,13 +55,10 @@ type Exporter struct {
 
 	couchLog *prometheus.GaugeVec
 
-	fabricWorker            *prometheus.GaugeVec
-	fabricOpenShard         *prometheus.GaugeVec
-	fabricReadRepairs       *prometheus.GaugeVec
-	fabricFailure           *prometheus.GaugeVec
```
Comment (gesellix) [inline lib/exporter.go]: Failure, MismatchedErrors, WriteQuorumErrors have been removed. That looks unrelated to the pull request. Could it be that I didn't notice the fields not being used when reviewing the previous pull request? 😅

File: lib/collector-v2.go [hunk 2]
```diff
@@ -20,7 +20,23 @@ func (e *Exporter) collectV2(stats Stats, exposedHttpStatusCodes []string, colle
...
+		e.requestTime.WithLabelValues(name, "Max").Set(nodeStats.Couchdb.RequestTime.Value.Max)
+		e.requestTime.WithLabelValues(name, "ArithmeticMean").Set(nodeStats.Couchdb.RequestTime.Value.ArithmeticMean)
+		e.requestTime.WithLabelValues(name, "GeometricMean").Set(nodeStats.Couchdb.RequestTime.Value.GeometricMean)
+		e.requestTime.WithLabelValues(name, "HarmonicMean").Set(nodeStats.Couchdb.RequestTime.Value.HarmonicMean)
+		e.requestTime.WithLabelValues(name, "Median").Set(nodeStats.Couchdb.RequestTime.Value.Median)
+		e.requestTime.WithLabelValues(name, "Variance").Set(nodeStats.Couchdb.RequestTime.Value.Variance)
+		e.requestTime.WithLabelValues(name, "StandardDeviation").Set(nodeStats.Couchdb.RequestTime.Value.StandardDeviation)
+		e.requestTime.WithLabelValues(name, "Skewness").Set(nodeStats.Couchdb.RequestTime.Value.Skewness)
+		e.requestTime.WithLabelValues(name, "Kurtosis").Set(nodeStats.Couchdb.RequestTime.Value.Kurtosis)
+		e.requestTime.WithLabelValues(name, "p50").Set(nodeStats.Couchdb.RequestTime.Value.Percentile[0][1])
```
[the remark this answers is not in the document]
Comment (kobuskc) [reply lib/collector-v2.go]: From what I can see it is created as a histogram here: https://github.com/apache/couchdb/blob/103a0624f309ea0d796176a55eb5faea68f26047/src/couch_stats/src/couch_stats_httpd.erl#L49

I haven't found any user configurable properties and it looks like the only way a user would be able to customize the metric would be to define a new one as described here: https://github.com/apache/couchdb/tree/master/src/couch_stats

With that said, it should be pretty straight forward to replace lines 34 - 39 with something like this to make it more resilient:

for _, percentile := nodeStats.Couchdb.RequestTime.Value.Percentile {
			e.requestTime.WithLabelValues(name, percentile).Set(nodeStats.Couchdb.RequestTime.Value.Percentile[percentile][1])
		}

If that's more acceptable, I'd be happy to make the change.

File: lib/exporter.go (hunk 1 above)
Comment (kobuskc) [reply lib/exporter.go]: Please accept my apologies. I created these previously but ended up not using them and I neglected to clean up properly. I should have picked it up before I made the previous pull request.

File: lib/collector-v2.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gesellix) [reply lib/collector-v2.go]: That would be great, please change it to be more resilient to config changes. Thanks!

File: lib/exporter.go (hunk 1 above)
Comment (gesellix) [reply lib/exporter.go]: No problem, me as a reviewer hasn't seen it either in the previous PR :)
And I'm fine with removing them here.

File: lib/collector-v2.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kobuskc) [reply lib/collector-v2.go]: I have made the change. The percentile metrics emitted by the loop will resemble the following:

couchdb_httpd_request_time{metric="50",node_name="couchdb@COUCH-1"} 10003.417
couchdb_httpd_request_time{metric="50",node_name="couchdb@COUCH-2"} 4.381
couchdb_httpd_request_time{metric="50",node_name="couchdb@COUCH-3"} 10003.379
couchdb_httpd_request_time{metric="75",node_name="couchdb@COUCH-1"} 10004.212
couchdb_httpd_request_time{metric="75",node_name="couchdb@COUCH-2"} 10003.362
couchdb_httpd_request_time{metric="75",node_name="couchdb@COUCH-3"} 10003.913
couchdb_httpd_request_time{metric="90",node_name="couchdb@COUCH-1"} 10004.281
couchdb_httpd_request_time{metric="90",node_name="couchdb@COUCH-2"} 10003.874
couchdb_httpd_request_time{metric="90",node_name="couchdb@COUCH-3"} 10004.226
couchdb_httpd_request_time{metric="95",node_name="couchdb@COUCH-1"} 10005.737
couchdb_httpd_request_time{metric="95",node_name="couchdb@COUCH-2"} 10003.902
couchdb_httpd_request_time{metric="95",node_name="couchdb@COUCH-3"} 10004.327
couchdb_httpd_request_time{metric="99",node_name="couchdb@COUCH-1"} 10005.737
couchdb_httpd_request_time{metric="99",node_name="couchdb@COUCH-2"} 10004.186
couchdb_httpd_request_time{metric="99",node_name="couchdb@COUCH-3"} 10004.327
couchdb_httpd_request_time{metric="999",node_name="couchdb@COUCH-1"} 10005.737
couchdb_httpd_request_time{metric="999",node_name="couchdb@COUCH-2"} 10004.186
couchdb_httpd_request_time{metric="999",node_name="couchdb@COUCH-3"} 10004.327

File: lib/collector.go [hunk 3]
```diff
@@ -68,6 +68,24 @@ var (
 		"owner_crashes",
 		"worker_crashes",
 		"closes"}
+
+	exposedRequestTimes = []string{
```
Comment (gesellix) [inline lib/collector.go]: The `exposedRequestTimes` and `exposedPercentiles` variables aren't used anywhere - or did I miss something?

Comment (kobuskc) [reply lib/collector.go]: Good catch. I have removed these variables.

**Comment by gesellix** [comment]
The release has been tagged with `24`, the Docker Image will be available with the same label (it's building right now and should be pushed in a few minutes).

**Pull request merged**